src/System.Net.HttpListener/src/System/Net/HttpListenerRequest.cs

```diff
@@ -46,6 +47,24 @@ public string[] UserLanguages
             }
         }
 
+        private static MethodInfo s_internalAddMethod = null;
+        private static MethodInfo InternalAddMethod
+        {
+            get
+            {
+                if (s_internalAddMethod == null)
+                {
+                    // We need to use CookieCollection.InternalAdd, as this method performs no validation on the Cookies.
+                    // Unfortunately this API is internal so we use reflection to access it. The method is cached for performance reasons.
+                    MethodInfo method = typeof(CookieCollection).GetMethod("InternalAdd", BindingFlags.NonPublic | BindingFlags.Instance);
+                    Debug.Assert(method != null, "We need to use an internal method named InternalAdd that is declared on Cookie.");
+                    s_internalAddMethod = method;
+                }
+
+                return s_internalAddMethod;
+            }
+        }
```

Comment:
Could you please clarify this @stephentoub? I'm trying to use `Delegate.CreateDelegate` or `MethodInfo.CreateDelegate` but they require us to know the instance to invoke Cookie.InternalSetMethod on